src/bokeh/core/property/constraints.py

```diff
@@ -0,0 +1,112 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2023, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+"""
+
+"""
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Standard library imports
+from typing import Any, TypeVar
+
+# Bokeh imports
+from ._sphinx import property_link, register_type_link, type_link
+from .bases import (
+    Init,
+    Property,
+    SingleParameterizedProperty,
+    TypeOrInst,
+)
+from .singletons import Intrinsic
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    "TypeOfAttr",
+)
+
+T = TypeVar("T")
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+class TypeOfAttr(SingleParameterizedProperty[T]):
+    """ Allows to check if an attribute of an object satisfies the
+        given type or a collection of types.
```

Comment:
This would be typical indentation
```suggestion
    """ Allows to check if an attribute of an object satisfies the
    given type or a collection of types.
```